File: benchmarks/oopsla25/subjects/411.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f411(const char *s) {
    int i = 0;
    while (i < strlen(s)) {
        if (s[i] == 'a') {
            i += 1;
        } else {
            int j = 0;
            while (j < strlen(s) - i) {
                if (s[i + j] == 'b') {
                    j += 1;
                }
                break;
            }
            assert(j > 0);
            i += j;
        }
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f411(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/411.py
import sys

def f411(s: str):
  i = 0
  while i < len(s):
    if s[i] == "a":
      i += 1
    else:
      j = 0
      while j < len(s)-i:
        if s[i+j] == "b":
          j += 1
        break
      assert j > 0
      i += j

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f411(my_string)
